src/main/java/jchess/gamelogic/Game.java

```diff
@@ -201,7 +204,47 @@ public boolean tryCheckRule()
 		}
 		return false;
 	}
+	
+	/**
+	 * Checking for Stalemate Rule
+	 * 
+	 * Stalemate is a situation in the game of chess where the player whose turn 
+	 * it is to move is not in check but has no legal move. 
+	 * The rules of chess provide that when stalemate occurs, the game ends as a draw.
+	 */
+	public void checkStalemate() {
+		m_oLogger.writeLog(LogLevel.INFO, "Instantiating check for Stalemate.", "checkStalemate", "Game");
 		
+		Boolean bstalemate = true;
+		IBoardAgent oCurrentBoard =  this.m_oGameModel.getBoard();
+		IPlayerAgent oCurrentPlayer = this.m_oGameState.getActivePlayer();
+		int icounter1 =0;
```

Comment:
Name should be nCounter1, or if instead of '1' if you could add some meaningful word. E.g. nCounterInner, nCounterPosition.

Reply:
Have changed the counter names to nCountNoOfPieces & nCountPiecesTrapped